pg_search/src/index/reader/index.rs

```diff
@@ -689,14 +701,24 @@ impl SearchIndexReader {
                     offset,
                 )
                 .into_iter()
-                .map(|((f, _, _), doc)| (f, doc))
+                .map(|((f, erased1, erased2), doc)| {
+                    let maybe_score = erased_features.try_get_score(&[erased1, erased2]);
+                    ((f, maybe_score), doc)
+                })
                 .collect()
             }
             x => {
-                panic!(
-                    "Unsupported sort-field count: {}. At most {MAX_TOPN_FEATURES} are supported.",
-                    x + 1,
-                )
+                if erased_features.score_index() == Some(x - 1) {
+                    panic!(
+                        "Unsupported sort-field count: {}. At most {} are supported when `paradedb.score` is requested.",
+                        x, MAX_TOPN_FEATURES - 1
+                    )
+                } else {
+                    panic!(
+                        "Unsupported sort-field count: {}. At most {MAX_TOPN_FEATURES} are supported.",
+                        x + 1,
+                    )
+                }
             }
```

Comment:
There is a corresponding check in https://github.com/paradedb/paradedb/blob/f3a3d940954ec93ec173e160fc467981e28742d9/pg_search/src/postgres/customscan/pdbscan/mod.rs#L1358-L1373 which should be updated to ensure that we don't propose `TopN` in this case.

Sorry: I should have left a comment here.
```suggestion
                // When changing the number of features supported here, ensure that `pullup_topn_pathkeys`
                // is updated as well.
                if erased_features.score_index() == Some(x - 1) {
                    panic!(
                        "Unsupported sort-field count: {}. At most {} are supported when `paradedb.score` is requested.",
                        x, MAX_TOPN_FEATURES - 1
                    )
                } else {
                    panic!(
                        "Unsupported sort-field count: {}. At most {MAX_TOPN_FEATURES} are supported.",
                        x + 1,
                    )
                }
            }
```

Reply:
FWIW: I think that that check is the reason why https://github.com/paradedb/paradedb/issues/3148 doesn't get TopN... I don't think it's related to parallelism.